src/BEditaClient.php

```diff
@@ -306,32 +306,52 @@ public function removeRelated($id, string $type, string $relation, array $data,
         return $this->delete(sprintf('/%s/%s/relationships/%s', $type, $id, $relation), json_encode($body), $headers);
     }
 
+    /**
+     * Create an array of objects with just id and type and filter objects with meta data.
+     *
+     * @param array $data Related resources or objects to insert
+     * @return array Mapped and filtered items
+     */
+    private function mapItemsAndMeta(array $data): array
+    {
+        $items = $data;
+        $withMeta = null;
+        if (!empty($data[0])) {
+            $items = array_map(function ($item) {
+                return [
+                    'id' => $item['id'],
+                    'type' => $item['type'],
+                ];
+            }, $data);
+            $withMeta = array_filter($data, function ($item) {
+                return !empty($item['meta']);
+            });
+        } elseif (!empty($data['meta'])) {
+            $withMeta = $data;
+        }
+
+        return compact('items', 'withMeta');
+    }
+
     /**
      * Replace a list of related resources or objects: previuosly related are removed and replaced with these.
      *
      * @param int|string $id Object id
      * @param string $type Object type name
      * @param string $relation Relation name
-     * @param string $data Related resources or objects to insert
+     * @param array $data Related resources or objects to insert
      * @param array|null $headers Custom request headers
      * @return array|null Response in array format
      */
     public function replaceRelated($id, string $type, string $relation, array $data, ?array $headers = null): ?array
     {
-        $items = array_map(function ($item) {
-            return [
-                'id' => $item['id'],
-                'type' => $item['type'],
-            ];
-        }, $data);
+        extract($this->mapItemsAndMeta($data));
```

Comment:
really don't like `extract` but it's only a matter of taste :wink:

Reply:
I suggested `extract`... maybe we can use a more esplicit statements